app/filters/verify_session.rb

```diff
@@ -1,5 +1,12 @@
 class VerifySession
   def self.before(controller)
+    if !allowed_pages?(controller) && controller.flash[:expired_session].present?
+      controller.reset_session
+      controller.redirect_to '/session/expired'
+    end
+    if !allowed_pages?(controller) && controller.flash[:confirmation].present?
+      controller.flash[:expired_session] = 'Session has expired'
```

Comment:
Same comment re: content might have ideas on this text. Also it should probably be retrieved from the en.yml config through the i18n module so we can internationalise this easily if we need to.

Reply:
idem